src/sms.rs

```diff
@@ -0,0 +1,201 @@
+                        .map_err(|_| Error::BufferTooSmall(None))?;
+                }
+            }
+        }
+
+        Ok(encoded_message)
+    }
+    // Pack a GSM 7 bit strings into 7 bites without 1 bit padding
+    // Reimplement from https://github.com/nrfconnect/sdk-nrf/blob/main/lib/sms/string_conversion.c#L294
+    fn pack_gsm7bit<const N: usize>(text: String<N>) -> Vec<u8, N> {
+        let mut src: usize = 0;
+        let mut dst: usize = 0;
+        let mut shift: usize = 0;
+        let len = text.len();
+        let mut bytes = text.into_bytes();
+
+        while src < len {
+            bytes[dst] = bytes[src] >> shift;
+            src += 1;
+            if src < len {
+                bytes[dst] |= bytes[src] << (7 - shift);
+                shift += 1;
+                if shift == 7 {
+                    shift = 0;
+                    src += 1;
+                }
+            }
+            dst += 1;
+        }
+        bytes.truncate(dst);
+        bytes
+    }
+    /// Sends the craftes message
+    /// `N` is need to provide internal buffer size for message and number encoding. Needs to be at least 2 * message.len() + 34
+    pub async fn send<const N: usize>(self) -> Result<(), Error> {
+        let encoded_number = Self::encode_number(self.number)?;
+
+        let encoded_message = Self::pack_gsm7bit(Self::ascii_to_gsm7bit::<N>(self.message)?);
+
+        let size = 2 + /* First header byte and TP-MR fields */
+		1 + /* Length of phone number */
+		1 + /* Phone number Type-of-Address byte */
+		encoded_number.len()/2 +
+		2 + /* TP-PID and TP-DCS fields */
+		1 + /* TP-UDL field */
+		encoded_message.len();
+
+        let mut at_cmgs: String<N> = String::new();
+        let mut encoded_number_len = encoded_number.len();
+        if self.number.trim_start_matches('+').len() % 2 != 0 {
+            encoded_number_len -= 1;
+        }
+        // Write the at command and begin with encoded number and it's lenght
+        write!(
+            &mut at_cmgs,
+            "AT+CMGS={}\r{:04X}{:04X}91{}",
+            size, 0x01, encoded_number_len, encoded_number
+        )
+        .map_err(|_| Error::BufferTooSmall(None))?;
+        // Write the message lenght
+        write!(&mut at_cmgs, "00{:04X}", self.message.len())
+            .map_err(|_| Error::BufferTooSmall(None))?;
+        // Write the GSM 7 bit packaged message as hex string
+        for c in &encoded_message {
+            write!(&mut at_cmgs, "{:02X}", c).map_err(|_| Error::BufferTooSmall(None))?;
+        }
+        // End character
+        write!(&mut at_cmgs, "\x1A").map_err(|_| Error::BufferTooSmall(None))?;
+
+        // Wait for LteLink to send the message
+        let lte_link = LteLink::new().await?;
+        lte_link.wait_for_link().await?;
+
+        // Configure the SMS parameters in modem
+        // This might need some rework when reciving SMS is add and reporting
+        if send_at::<64>("AT+CNMI=3,2,0,1").await?.as_str() != "OK\r\n" {
```

Comment:
You can decrease this 64 number to the size of the message that you actually expect to get back. Saves some stackspace